src/Nest/QueryDsl/MultiTermQueryRewrite/RewriteMultiTerm.cs

```diff
@@ -46,10 +50,191 @@ public enum RewriteMultiTerm
+		/// </summary>
+		public static MultiTermQueryRewrite ConstantScoreBoolean { get; } = new MultiTermQueryRewrite(RewriteMultiTerm.ConstantScoreBoolean);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into should clause in boolean query, and keeps the scores
+		/// as computed by the query. This rewrite method only uses the top scoring terms so it will not overflow boolean
+		///  max clause count. <param name="size" /> controls the size of the top scoring terms to use.
+		/// </summary>
+		public static MultiTermQueryRewrite TopTerms(int size) => new MultiTermQueryRewrite(RewriteMultiTerm.TopTermsN, size);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into should clause in boolean query, but the scores are only
+		/// computed as the boost. This rewrite method only uses the top scoring terms so it will not overflow the boolean
+		///  max clause count. <param name="size" /> controls the size of the top scoring terms to use.
+		/// </summary>
+		public static MultiTermQueryRewrite TopTermsBoost(int size) => new MultiTermQueryRewrite(RewriteMultiTerm.TopTermsBoostN, size);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into should clause in boolean query, but all term queries compute
+		///  scores as if they had the same frequency. In practice the frequency which is used is the maximum frequency of all
+		///  matching terms. This rewrite method only uses the top scoring terms so it will not overflow boolean max clause count.
+		/// <param name="size" /> controls the size of the top scoring terms to use.
+		/// </summary>
+		public static MultiTermQueryRewrite TopTermsBlendedFreqs(int size) => new MultiTermQueryRewrite(RewriteMultiTerm.TopTermsBlendedFreqsN, size);
+
+		internal static MultiTermQueryRewrite Create(string value)
+		{
+			if (value == null) throw new ArgumentNullException(nameof(value));
+
+			string rewriteType;
+			var size = 0;
+			var firstDigitIndex = value.IndexOfAny(DigitCharacters);
+
+			if (firstDigitIndex > -1)
+			{
+				rewriteType = value.Substring(0, firstDigitIndex) + "N";
+				size = int.Parse(value.Substring(firstDigitIndex));
+			}
+			else
+				rewriteType = value;
+
+			var rewriteMultiTerm = rewriteType.ToEnum<RewriteMultiTerm>();
+			if (rewriteMultiTerm == null)
+				throw new InvalidOperationException($"Unsupported {nameof(RewriteMultiTerm)} value: '{rewriteType}'");
+
+			switch (rewriteMultiTerm)
+			{
+				case RewriteMultiTerm.ConstantScore:
+					return ConstantScore;
+				case RewriteMultiTerm.ScoringBoolean:
+					return ScoringBoolean;
+				case RewriteMultiTerm.ConstantScoreBoolean:
+					return ConstantScoreBoolean;
+				case RewriteMultiTerm.TopTermsN:
+					return TopTerms(size);
+				case RewriteMultiTerm.TopTermsBoostN:
+					return TopTermsBoost(size);
+				case RewriteMultiTerm.TopTermsBlendedFreqsN:
+					return TopTermsBlendedFreqs(size);
+				default:
+					throw new ArgumentOutOfRangeException();
```

Comment:
Not sure this is the right exception type to throw here.  Regardless though, we should add a message and include the term that didn't match.